subsys/shell/shell.c

```diff
@@ -1740,3 +1748,21 @@ int shell_execute_cmd(const struct shell *shell, const char *cmd)
 
 	return shell_execute(shell);
 }
+
+int shell_set_flags(const struct shell *shell, u32_t flag)
```

Comment:
Having runtime configuration for this is nice but this is a bit cumbersome to use. Would it be possible to have a init function for a given shell instance so that these flags could be set before user actually uses the shell.
For example something like `void shell_instance_init(struct shell *shell);`

Reply:
Hmm, I meant actually that user could define a init function for shell_init() which would be called before the shell is actually used.

Reply:
@jukkar : I can use macro: `SHELL_DEFINE` for this purpose.
Shall I remove in such case `shell_set_flags` function?

Reply:
Using SHELL_DEFINE() might make more sense here. Also using global Kconfig option would affect all shells globally which might not be desirable.

Reply:
I wonder what is your opinion of the shell instance init function. For example I have the zperf shell in networking side where I need to execute an init function which would do some init of the system and print some output to shell.
Currently I have this kind of pattern there.

```
static void do_init(const struct shell *shell)
{
	static bool init_ok;

	if (!init_ok) {
		zperf_init(shell);
		init_ok = true;
	}
}

static int cmd_xyz(const struct shell *shell, size_t argc, char *argv[])
{
	do_init(shell);
...
}
```

In all the command handlers I call this `do_init(shell)` which is a bit cumbersome but certainly doable. If I could specify an init function in `shell_init()`, it would make the code a bit simpler there.

Reply:
@jukkar : done

Regarding init function I need to think about it during the weekend.